apps/server/src/modules/tool/controller/dto/request/external-tool-create.params.ts

```diff
@@ -0,0 +1,61 @@
+import { IsBoolean, IsNumber, IsOptional, IsString, ValidateNested } from 'class-validator';
+import { ApiExtraModels, ApiProperty, ApiPropertyOptional, getSchemaPath } from '@nestjs/swagger';
+import { CustomParameterCreateParams } from '@src/modules/tool/controller/dto/request/custom-parameter.params';
+import { BasicToolConfigParams } from '@src/modules/tool/controller/dto/request/basic-tool-config.params';
+import { Lti11ToolConfigParams } from '@src/modules/tool/controller/dto/request/lti11-tool-config.params';
+import { Oauth2ToolConfigParams } from '@src/modules/tool/controller/dto/request/oauth2-tool-config.params';
+import { Type } from 'class-transformer';
+import { ExternalToolConfigCreateParams } from '@src/modules/tool/controller/dto/request/external-tool-config.params';
+import { ToolConfigType } from '@src/modules/tool/interface/tool-config-type.enum';
+
+@ApiExtraModels(Lti11ToolConfigParams, Oauth2ToolConfigParams, BasicToolConfigParams)
+export class ExternalToolParams {
+	@IsString()
+	@ApiProperty()
+	name!: string;
+
+	@IsString()
+	@ApiProperty()
+	url!: string;
```

Comment:
this is an optional field in the documentation. I think we should talk with Alex what the difference between 'url' and 'baseUrl' is and which ones are required.